src/Type/Regex/RegexExpressionHelper.php

```diff
@@ -88,13 +91,24 @@ public function getPatternModifiers(string $pattern): ?string
 
 	public function removeDelimitersAndModifiers(string $pattern): string
 	{
+		$pattern = ltrim($pattern);
+
 		$endDelimiterPos = $this->getEndDelimiterPos($pattern);
 
 		if ($endDelimiterPos === false) {
 			return $pattern;
 		}
 
-		return substr($pattern, 1, $endDelimiterPos - 1);
+		$raw = substr($pattern, 1, $endDelimiterPos - 1);
+
+		$startDelimiter = substr($pattern, 0, 1);
+		$endDelimiter = $this->getPatternDelimiter($pattern);
+		assert($endDelimiter !== null);
+
+		$raw = preg_replace('~(?<!\\\)(\\\\\\\)*\K\\\(?=[' . preg_quote($startDelimiter . $endDelimiter, '~') . '])~', '', $raw);
+		assert($raw !== null);
```

Comment:
@staabm can you please review?

Reply:
why is this ugly regex needed? what does it do?

Reply:
The regex is beautiful :), it strips escaping `\` before delimiter if not preceded by another (escaped) `\`.

Reply:
@Seldaek do you have an idea whether this can be achieved in a more readable way?

Reply:
The regex is not longer needed, regexes are parsed differently.